src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts

```diff
@@ -152,6 +209,38 @@ class ZHAConfigDashboard extends LitElement {
     this._configuration = await fetchZHAConfiguration(this.hass!);
   }
 
+  private async _fetchSettings(): Promise<void> {
+    this._networkSettings = await fetchZHANetworkSettings(this.hass!);
+  }
+
+  private async _createAndDownloadBackup(): Promise<void> {
+    this._generatingBackup = true;
+    const backup_and_metadata: ZHANetworkBackupAndMetadata =
+      await createZHANetworkBackup(this.hass!);
+    this._generatingBackup = false;
+
+    if (!backup_and_metadata.is_complete) {
+      await showAlertDialog(this, {
+        title: "Backup is incomplete",
+        text: "A backup has been created but it is incomplete and cannot be restored. This is a coordinator firmware limitation.",
+      });
+    }
+
+    const backupJSON: string =
+      "data:text/plain;charset=utf-8," +
+      encodeURIComponent(JSON.stringify(backup_and_metadata.backup, null, 4));
+    const backupTime: Date = new Date(
+      Date.parse(backup_and_metadata.backup.backup_time)
+    );
+    let filename: string = backupTime.toISOString().replace(/:/g, "-");
+
+    if (!backup_and_metadata.is_complete) {
+      filename += ", incomplete";
```

Comment:
comma in filename is weird. Maybe prefix name with `INCOMPLETE_` or maybe even prevent downloading?